AspNetCore.ReCaptcha/ReCaptchaGenerator.cs

```diff
@@ -92,13 +92,11 @@ public static IHtmlContent ReCaptchaV2Invisible(Uri baseUrl, string siteKey, str
 
         public static IHtmlContent ReCaptchaV3(Uri baseUrl, string siteKey, string action, string language, string callBack, int id, string nonce = null)
         {
+            var nonceAttribute = string.IsNullOrEmpty(nonce) ? string.Empty : @$" nonce=""{nonce}""";
             var content = new HtmlContentBuilder();
             content.AppendHtml(@$"<input id=""g-recaptcha-response-{id}"" name=""g-recaptcha-response"" type=""hidden"" value="""" />");
-            content.AppendFormat(@"<script src=""{0}api.js?render={1}&hl={2}""></script>", baseUrl, siteKey, language);
-            content.AppendHtml("<script");
-            if (!string.IsNullOrEmpty(nonce))
-                content.AppendFormat(" nonce=\"{0}\"", nonce);
-            content.AppendHtml(">");
+            content.AppendHtml(@$"<script src=""{baseUrl}api.js?render={siteKey}&hl={language}""{nonceAttribute}></script>");
```

Comment:
We explicitly used `AppendFormat` here as it encodes the parameters so they're safe for injecting into the html code. If we change it to `AppendHtml` that encoding is assumed to be done beforehand, however as these parameters are provided to us we can't assume they are encoded. Although the uri, sitekey and language used here are most-likely not user provided, I think it's still safer to use AppendFormat here instead.

```suggestion
            content.AppendFormat(@"<script src=""{0}api.js?render={1}&hl={2}"", baseUrl, siteKey, language);
            if (!string.IsNullOrEmpty(nonce))
                content.AppendFormat(" nonce=\"{0}\"", nonce);
            content.AppendHtml("></script>");
```

Reply:
See updated changes with AddNonce extension method